core/src/main/java/hudson/model/Run.java

```diff
@@ -2302,6 +2302,9 @@ public EnvVars getEnvironment() throws IOException, InterruptedException {
         for (EnvironmentContributor ec : EnvironmentContributor.all().reverseView())
             ec.buildEnvironmentFor(this,env,listener);
 
+        for (EnvironmentContributingAction a : getActions(EnvironmentContributingAction.class))
+            a.buildEnvVars(this,env,n);
```

Comment:
Nice 2 have for new extension invocations: try/catch unexpected runtime exceptions. NIT in this case since there is same old code above